src/main/java/mate/academy/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.dao;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import mate.academy.db.Storage;
+import mate.academy.lib.Dao;
+import mate.academy.model.Manufacturer;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        if (manufacturer != null) {
+            Storage.addManufacturer(manufacturer);
+        }
+        return manufacturer == null ? null
+                : Storage.manufacturers.get(Storage.manufacturers.size() - 1);
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        return Storage.manufacturers.stream()
+                .filter(m -> Objects.equals(id, m.getId()))
+                .findFirst();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Manufacturer oldValue = get(manufacturer.getId()).orElseThrow();
+        Storage.manufacturers.remove(oldValue);
+        Storage.manufacturers.add(manufacturer);
+        return oldValue;
```

Comment:
in this case you'll add manufacture to the end of the list, but you need to update current manufacturer
According to Roman: "we should iterate the list to find the needed object"